sql/wsrep_mysqld.cc

```diff
@@ -1558,7 +1559,9 @@ static bool wsrep_prepare_key_for_isolation(const char *db, const char *table,
     }
     default:
       assert(0);
-      WSREP_ERROR("Unsupported protocol version: %ld", wsrep_protocol_version);
+      WSREP_ERROR("Unsupported protocol version: %ld",
+                  static_cast<std::underlying_type<wsrep_version>::type>(
```

Comment:
This static_cast is really needed?

Reply:
Since enum classes are NOT implicitly convertible to int, this seemed like a good solution to fetch the underlying type.
Also, if the type changes in the future, this wouldn't require a change.

Reply:
You are right, enum class adds explicit scoping but removes implicit conversion, which results in the need for such an unpleasant casting...

@dutow @VarunNagaraju What about the following approach?
```
#include <iostream>

namespace MyEnum {
enum MyEnum : long {
  one = 1,
  two = 2
};
}

int main() {
  std::cout << MyEnum::one << "\n";
  return 0;
}
```

Reply:
It removes the unpleasant casting and the members are scoped. Seems good to me.